src/marks/axis.js

```diff
@@ -505,56 +505,61 @@ function labelOptions(
+            // scale.ticks because it’s supported by d3.utcTicks; but
+            // quantitative scales and d3.ticks do not support numeric
+            // intervals for scale.ticks, so we compute them here.
+            const [min, max] = extent(scale.domain());
+            data = interval.range(min, interval.offset(interval.floor(max))); // inclusive max
           } else {
-            interval = maybeRangeInterval(interval === undefined ? scale.interval : interval, scale.type);
-            if (interval !== undefined) {
-              // For time scales, we could pass the interval directly to
-              // scale.ticks because it’s supported by d3.utcTicks; but
-              // quantitative scales and d3.ticks do not support numeric
-              // intervals for scale.ticks, so we compute them here.
-              const [min, max] = extent(scale.domain());
-              data = interval.range(min, interval.offset(interval.floor(max))); // inclusive max
-            } else {
-              const [min, max] = extent(scale.range());
-              ticks = (max - min) / (tickSpacing === undefined ? (k === "x" ? 80 : 35) : tickSpacing);
-              data = scale.ticks(ticks);
-            }
+            const [min, max] = extent(scale.range());
+            ticks = (max - min) / (tickSpacing === undefined ? (k === "x" ? 80 : 35) : tickSpacing);
+            data = scale.ticks(ticks);
           }
-        } else {
-          data = scale.domain();
-        }
-        if (k === "y" || k === "x") {
-          facets = [range(data)];
-        } else {
-          channels[k] = {scale: k, value: identity};
         }
+      } else {
+        data = scale.domain();
       }
-      initialize?.call(this, scale, ticks, channels);
-      const initializedChannels = Object.fromEntries(
-        Object.entries(channels).map(([name, channel]) => {
-          return [name, {...channel, value: valueof(data, channel.value)}];
-        })
-      );
-      if (initializeFacets) facets = context.filterFacets(data, initializedChannels);
-      return {data, facets, channels: initializedChannels};
-    })
-  );
+      if (k === "y" || k === "x") {
+        facets = [range(data)];
+      } else {
+        channels[k] = {scale: k, value: identity};
+      }
+    }
+    initialize?.call(this, scale, ticks, channels);
+    const initializedChannels = Object.fromEntries(
+      Object.entries(channels).map(([name, channel]) => {
+        return [name, {...channel, value: valueof(data, channel.value)}];
+      })
+    );
+    if (initializeFacets) facets = context.filterFacets(data, initializedChannels);
+    return {data, facets, channels: initializedChannels};
+  });
+
+  // Apply the filter option after the initializer has determined the mark’s data.
+  if (f != null) i = filter(f, i);
+  if (s != null) throw new Error("Unsupported axis option: sort");
+  if (r != null) throw new Error("Unsupported axis option: reverse");
```

Comment:
Match the style of our other errors:

```suggestion
  if (s != null) throw new Error("unsupported axis option: sort");
  if (r != null) throw new Error("unsupported axis option: reverse");
```

Reply:
The version with composeInitializer works (including sort+reverse!); I'm still trying to make the second version work.

Reply:
(Couldn't make the second version work.)

Reply:
Yep, I got the order reversed in my suggestion. I’ll take one more look but I think the composeInitializer approach is fine too! Thanks.

Reply:
Got it working. One call to construct the “basic” initializer (which could even be a custom **initializer**!) and a second to compose that with the “axis” initializer.

Reply:
much better! this consistency will hopefully unlock more things in the future (such as zooming)